File: src/routes/examples/norns/notes/[id]/+page.server.c
import { notes } from '$lib/norns/notes/server/public'
import { updateNoteSchema } from '$lib/norns/notes/shared/schema'

export load := page.load
	handler: async ({ container, params }) =>
		id := Number params.id
		note := await notes(container).get id
		throw error 404, 'note not found' unless note
		{ note }

export actions := page.actions
	update:
		input: updateNoteSchema
		run: async ({ input, container, event }) =>
			id := Number event.params.id
			await notes(container).update id, input
			{ saved: true }

	delete:
		run: async ({ container, event }) =>
			await notes(container).remove Number(event.params.id)
			throw redirect 303, '/examples/norns/notes'

File: src/routes/examples/svelte/notes/[id]/+page.server.ts
import { error, fail, redirect } from '@sveltejs/kit';
import * as v from 'valibot';
import type { Actions, PageServerLoad } from './$types';
import * as notes from '$lib/svelte/notes/service';
import { db } from '$lib/svelte/notes/db';
import { updateNoteSchema } from '$lib/svelte/notes/schema';

export const load: PageServerLoad = async ({ params, platform }) => {
	const id = Number(params.id);
	const note = await notes.get(db(platform), id);
	if (!note) throw error(404, 'note not found');
	return { note };
};

export const actions: Actions = {
	update: async ({ request, params, platform }) => {
		const id = Number(params.id);
		const data = await request.formData();
		const result = v.safeParse(updateNoteSchema, {
			title: data.get('title')?.toString(),
			body: data.get('body')?.toString() ?? ''
		});
		if (!result.success) return fail(400, { errors: result.issues });
		await notes.update(db(platform), id, result.output);
		return { saved: true };
	},

	delete: async ({ params, platform }) => {
		await notes.remove(db(platform), Number(params.id));
		throw redirect(303, '/examples/svelte/notes');
	}
};
